src/zkevm_specs/tx.py

```diff
@@ -0,0 +1,366 @@
+from .encoding import U8, is_circuit_code
+from typing import NamedTuple, Tuple, List, Sequence, Set
+from .util import FQ, RLC, U160, U256, U64
+from enum import IntEnum, auto
+from eth_keys import KeyAPI  # type: ignore
+import rlp  # type: ignore
+from eth_utils import keccak
+
+
+class Tag(IntEnum):
+    """
+    Tag used as second key in the Tx Circuit Rows to "select" the transaction field target.
+    Can be encoded in 4 bits.
+    """
+
+    Nonce = 1
+    Gas = 2
+    GasPrice = 3
+    GasTipCap = 4
+    GasFeeCap = 5
+    CallerAddress = 6
+    CalleeAddress = 7
+    IsCreate = 8
+    Value = 9
+    CallDataLength = 10
+    TxSignHash = 11
+    CallData = 12
+    Pad = 13
+
+
+class Row:
+    """
+    Tx circuit row
+    """
+
+    tx_id: FQ
+    tag: FQ
+    index: FQ
+    value: FQ
+
+    def __init__(self, tx_id: FQ, tag: FQ, index: FQ, value: FQ):
+        self.tx_id = tx_id
+        self.tag = tag
+        self.index = index
+        self.value = value
+
+
+class KeccakTable:
+    # The columns are: (is_enabled, input_rlc, input_len, output_rlc)
+    table: Set[Tuple[FQ, FQ, FQ, FQ]]
+
+    def __init__(self):
+        self.table = set()
+        self.table.add((FQ(0), FQ(0), FQ(0), FQ(0)))  # Add all 0s row
+
+    def add(self, input: bytes, randomness: FQ):
+        output = keccak(input)
+        self.table.add(
+            (FQ(1), RLC(input, randomness).value, FQ(len(input)), RLC(output, randomness).value)
+        )
```

Comment:
You're right!  Resolved in https://github.com/appliedzkp/zkevm-specs/pull/150/commits/a1792dd1077b01bb23cbf8618cb8882dea646fdb

For lines 191-193 I kept the construction of `pub_key_bytes` as before (so that it matches the expected byte representation of the public key), and instead applied the `reversed` in the call to the RLC.